SPECS/golang/golang-1.18.spec

```diff
@@ -0,0 +1,247 @@
+%global goroot          %{_libdir}/golang
+%global gopath          %{_datadir}/gocode
+%ifarch aarch64
+%global gohostarch      arm64
+%else
+%global gohostarch      amd64
+%endif
+%define debug_package %{nil}
+%define __strip /bin/true
+# rpmbuild magic to keep from having meta dependency on libc.so.6
+%define _use_internal_dependency_generator 0
+%define __find_requires %{nil}
+Summary:        Go
+Name:           golang
+Version:        1.18.8
+Release:        3%{?dist}
+License:        BSD-3-Clause
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          System Environment/Security
+URL:            https://golang.org
+Source0:        https://golang.org/dl/go%{version}.src.tar.gz
+Source1:        https://dl.google.com/go/go1.4-bootstrap-20171003.tar.gz
+Patch0:         go14_bootstrap_aarch64.patch
+# CVE-2022-41717 is fixed in 1.18.9
+Patch1:         CVE-2022-41717.patch
+Obsoletes:      %{name} < %{version}
+Provides:       %{name} = %{version}
```

Comment:
Isn't this the default behaviour for all packages? Or is one of the macro (`_use_internal_dependency_generator 0`?) messing with RPM's automation and thus this is required?

Reply:
I've taken the golang spec as it was before moving to 1.19 and just created a spec for version 1.18 from it (nothing changed)